src/app/InteractionModelEngine.cpp

```diff
@@ -227,6 +227,56 @@ void InteractionModelEngine::OnResponseTimeout(Messaging::ExchangeContext * ec)
     ChipLogProgress(DataManagement, "Time out! failed to receive echo response from Exchange: %d", ec->GetExchangeId());
 }
 
+void InteractionModelEngine::ReleaseClusterInfoListToPool(ReadHandler * const apReadHandler)
+{
+    ClusterInfo * const clusterInfoList = apReadHandler->GetClusterInfoList();
+    size_t numClusterInfos              = apReadHandler->GetNumClusterInfos();
+    size_t numClusterInfosToBeAffected;
+
+    apReadHandler->ClearClusterInfo();
+
+    if (numClusterInfos == 0)
+    {
+        ChipLogDetail(DataManagement, "No cluster instances allocated");
+        return;
+    }
+
+    // make sure everything is still sane
+    ChipLogIfFalse(clusterInfoList >= mClusterInfoPool);
+    ChipLogIfFalse(numClusterInfos <= mNumClusterInfos);
+
+    // mPathGroupPool + IM_PUBLISHER_MAX_NUM_PATH_GROUPS is a pointer which points to the last+1byte of this array
+    // clusterInfoList is a pointer to the first cluster instance to be released
+    // the result of subtraction is the number of cluster instances from clusterInfoList to the end of this array
+    numClusterInfosToBeAffected = ((size_t) mClusterInfoPool + mNumClusterInfos) - (size_t) clusterInfoList;
```

Comment:
use C++-style casting since that requires brackets and makes it clear what gets cast (especially in the plus case).

if clusterInfoList is a pointer, why is casting to size_t correct? Please do pointer differences and only then cast to count. I expect pointer differences to also include a sizeof() somewhere which these do not (and unless a cluster info is 1 byte, then this seems wrong).

This is "tricky code". please try to make it less tricky.

Reply:
yep, just update with c++ casting.